openhands/runtime/utils/async_bash.py

```diff
@@ -17,6 +19,13 @@ async def execute(command: str, work_dir: str) -> CommandResult:
         if not command:
             return CommandResult(content='', exit_code=0)
 
+        # Handle username similar to BashSession
+        if username in ['root', 'openhands']:
+            # Use sudo with bash -c to properly handle command substitution and special characters
+            # Replace single quotes with escaped double quotes to avoid issues with nested commands
+            escaped_command = command.replace("'", '\\"')
+            command = f'sudo -u {username} bash -c "{escaped_command}"'
```

Comment:
@openhands fix this logic. DO NOT use `-c` or anything else that requires escaping commands. Instead find a robust way to change how we e.g. create the subprocess to make sure the correct user is used

Reply:
@OpenHands fix this logic. DO NOT use -c or anything else that requires escaping commands. Instead find a robust way to change how we e.g. create the subprocess to make sure the correct user is used